### PR TITLE
Add HelloWorldMetal demo

### PR DESCRIPTION
**Pull request by hanton**
HelloWorld-Metal is a simple image classification application that demonstrates how to use PyTorch C++ libraries with Metal support.

### CODE REVIEW

**Comment by hanton** [comment]
Tested on pod LibTorch-Lite-Nightly `1.10.0.20210817`. cc: @xta0 @yuhc for review help

**Comment by xta0** [comment]
Can you rename the project to HelloWorld-Metal

File: HelloWorldMetal/HelloWorldMetal/Podfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+platform :ios, '12.0'
+target 'HelloWorldMetal' do
```
Comment (xta0) [inline HelloWorldMetal/HelloWorldMetal/Podfile:3]: HelloWorld-Metal

File: HelloWorldMetal/trace_model.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import torch
```
Comment (xta0) [inline HelloWorldMetal/trace_model.py:1]: Can we drop this file for now? After we get the MacOS nightly working, we can check in this script.

File: HelloWorldMetal/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- XCode 11.0 or above
+- iOS 12.0 or above
+
+## The HelloWorldMetal Example
+
+HelloWorldMetal is a simple image classification application that demonstrates how to use PyTorch C++ libraries with Metal support. The code is written in Swift and uses Objective-C as a bridge.
+
+We’ll be using the mobilenetv2 model as an example. Since the mobile GPU features are currently in the prototype stage, you’ll need to build a custom pytorch binary from source. For the time being, only a limited number of operators are supported, and certain client side APIs are subject to change in the future versions.
+
+### Model Preparation
```
Comment (xta0) [inline HelloWorldMetal/README.md:12]: I think we can just add a link to the OSS Metal tutorial.

File: HelloWorldMetal/HelloWorldMetal/Podfile (hunk 1 above)
Comment (hanton) [reply HelloWorldMetal/HelloWorldMetal/Podfile:3]: Fixed.

File: HelloWorldMetal/README.md (hunk 3 above)
Comment (hanton) [reply HelloWorldMetal/README.md:12]: Sonds good. A link to `(Prototype) Use iOS GPU in PyTorch` is added.

**Comment by xta0** [comment]
LGTM.

**Pull request merged**